spyderlib/widgets/file_switcher.py

```diff
@@ -0,0 +1,415 @@
+
+        self.tabs = tabs
+        self.tab_data = tab_data
+        self.filtered_index_to_path = None
+        self.full_index_to_path = None
+        self.path_to_line_count = None
+
+        edit_layout = QHBoxLayout()
+        self.edit = QLineEdit(self)
+        self.up_down_filter = UpDownFilter()
+        self.up_down_filter.up_down.connect(self.handle_up_down)
+        self.edit.installEventFilter(self.up_down_filter)
+        self.edit.returnPressed.connect(self.handle_edit_file)
+        self.edit.textChanged.connect(lambda text: self.synchronize(None))
+
+        edit_layout.addWidget(self.edit)
+        self.button_help = HelperToolButton()
+        self.button_help.setIcon(get_std_icon('MessageBoxInformation'))
+        style = """
+            QToolButton {
+              border: 1px solid grey;
+              padding:0px;
+              border-radius: 2px;
+              background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
+                  stop: 0 #f6f7fa, stop: 1 #dadbde);
+            }
+            """
+        self.button_help.setStyleSheet(style)
+        help_str = _("""
+            Press <b>Enter</b> to switch files.<br>
+            Or <b>Esc</b> to cancel.<br>
+            Type to filter filenames.<br>
+            Use <b>:number</b> to go to a line,<br>
+            e.g. 'main:42'.<br>
+            Press <b>Ctrl+W</b> to close current tab.
+            """)
+        self.button_help.setToolTip(help_str)
+        edit_layout.addWidget(self.button_help)
+
+        self.listwidget = QListWidget(self)
+        self.listwidget.setItemDelegate(HTMLDelegate(self))
+        self.listwidget.itemSelectionChanged\
+                       .connect(self.item_selection_changed)
+        self.listwidget.itemActivated.connect(self.handle_edit_file)
+        self.original_path = None
+        self.original_line_num = None
+        self.line_num = -1
+        self.line_num_modified_for_path = None
+
+        self.edit_line.connect(self.handle_edit_line)
+        self.rejected.connect(self.handle_rejection)
+        new_shortcut("Ctrl+W", self, lambda:
+                     self.close_file.emit(
+                         self.filtered_index_to_full(
+                             self.listwidget.currentRow())))
+
+        vlayout = QVBoxLayout()
+        vlayout.addLayout(edit_layout)
+        vlayout.addWidget(self.listwidget)
+        self.setLayout(vlayout)
+
+        self.edit.selectAll()
+        self.edit.setFocus()
+        geo = parent.geometry()
+        width = min(300, 0.8*geo.width())
+        self.listwidget.setMinimumWidth(width)
+
+        left = parent.geometry().width()/2 - width/2
+        top = 0
+        while parent:
+            geo = parent.geometry()
+            top += geo.top()
+            left += geo.left()
+            parent = parent.parent()
+        self.move(left,
+                  top + self.tabs.tabBar().geometry().height() + 1)
+        # Note: the +1px on the top is a hack
```

Comment:
Change this to something like `It just looks better with +1`. You have to explain the reason why it is added, confirming that it's strange is not really helpful :wink: